src/main/java/frc/robot/Controls.java

```diff
@@ -349,7 +350,16 @@ private void configureSuperStructureBindings() {
                           switch (scoringMode) {
                             case CORAL -> getCoralBranchHeightCommand();
                             case ALGAE -> Commands.sequence(
-                                    superStructure.algaeNetScore(driverController.rightBumper()),
+                                    BargeAlign.driveToBlackLine(s.drivebaseSubsystem),
+                                    BargeAlign.driveToBarge(s.drivebaseSubsystem)
```

Comment:
It seems inefficient and potentially risky to stop abruptly between these two commands and then start up again (as `BargeAlign.end()` in `driveToBlackLine` stops all motion). Ideally `driveToBlackLine` ends by setting the speed to `-xBargeDriveSpeed` or `xBargeDriveSpeed` so there's a smooth transition. Might be better to just combine `driveToBlackLine` and `driveToBarge` into a single composite command that manages that transition & safely stops the robot if it's interrupted.

Reply:
Good point about the abrupt stop!

I'm a bit iffy about doing the sequence logic ourselves inside the class command instead of via a sequential, but It'd probably be manageable. We would need to sort out how to cause the scoring sequence to start at the right time, though.

Reply:
Maybe pass in a command to run when the black line is reached to the BargeAlign command?

Reply:
We could make that work, but it'd make the logic harder to reason about. (Coordinating sequences and parallels of actions can't really be cleanly expressed by logic in `execute()`, and manually scheduling commands is a code smell.)

The cleanest solution is probably making a `Command BargeAlign.bargeAlign(CommandSwerveDrivetrain, SuperStructure)` method that encapsulates the logic of driving to the black line, then moving to the black line and scoring. That way, we can safely make `end()` not stop the drive base, since we can make the separate steps private and guarantee the drive to black line command will always be followed by the drive to barge command.